### PR TITLE
fix(rpc/trace): return empty if after >= traces

### LINKED ISSUE #11711: trace_filter returns random results
### Describe the bug

I am using `trace_filter` to check the txes to/from an account. Reth is returning results that do not involve that address in any way.

### Steps to reproduce

Query:

```terminal
curl -X POST --data '{"jsonrpc":"2.0","id":1,"method":"trace_filter","params":[{"fromBlock":"0x13f7ab5","toBlock":"0x13f7ad3","toAddress":["0x897a96Deb94291B785fAfF64088f7202A019f38e"],"after":100,"count":100}]}' -H 'Content-Type: application/json'
```

This should return an empty result as does Erigon. This is because of the `after` being set to 100.

### Node logs

_No response_

### Platform(s)

Linux (x86)

### What version/commit are you on?

I am not running the node myself. Is it possible to get this via API?

### What database version are you on?

I am not running the node myself. Is it possible to get this via API?

### Which chain / network are you on?

Ethereum mainnet

### What type of node are you running?

Archive (default)

### What prune config do you use, if any?

_No response_

### If you've built Reth from source, provide the full command you used

_No response_

### Code of Conduct

- [X] I agree to follow the Code of Conduct

### PR DESCRIPTION
**Pull request by jsvisa**
fix https://github.com/paradigmxyz/reth/issues/11711#issuecomment-2410798009

### CODE REVIEW

File: crates/rpc/rpc/src/trace.rs [hunk 1]
```diff
@@ -325,7 +325,9 @@ where
         // apply after and count to traces if specified, this allows for a pagination style.
         // only consider traces after
         if let Some(after) = after.map(|a| a as usize).filter(|a| *a < all_traces.len()) {
-            all_traces = all_traces.split_off(after);
+            all_traces.drain(0..after);
+        } else {
+            return Ok(vec![])
```
Comment (mattsse) [inline crates/rpc/rpc/src/trace.rs:330]: this now returns an empty vec if after is missing from the request?

this doesn't sound right?

could we also explain what `after` does exactly, not immediately obvious after reading the docs here

Comment (jsvisa) [reply crates/rpc/rpc/src/trace.rs:330]: sorry, forgot it.

Comment (jsvisa) [reply crates/rpc/rpc/src/trace.rs:330]: I'll need to read the og parity's codebase to see what's the usage of this field :)

Comment (mattsse) [reply crates/rpc/rpc/src/trace.rs:330]: lets follow erigon's impl here because the parity is no longer in use

File: crates/rpc/rpc/src/trace.rs [hunk 2]
```diff
@@ -322,13 +322,18 @@ where
...
-        // apply after and count to traces if specified, this allows for a pagination style.
-        // only consider traces after
-        if let Some(after) = after.map(|a| a as usize).filter(|a| *a < all_traces.len()) {
-            all_traces = all_traces.split_off(after);
+        // Skips the first `after` number of matching traces.
+        // If `after` is greater than or equal to the number of matched traces, it returns an empty
+        // array.
+        if let Some(after) = after.map(|a| a as usize) {
+            if after < all_traces.len() {
+                all_traces.drain(0..after);
```
Comment (mattsse) [inline crates/rpc/rpc/src/trace.rs:330]:
```suggestion
                all_traces.drain(..after);
```

Comment (mattsse) [inline crates/rpc/rpc/src/trace.rs:330]: this is the same as before but adds the return, which is the fix?

**Review by mattsse — APPROVED** [review]

**Pull request merged**